tests/conftest.py

```diff
@@ -3143,12 +3156,18 @@ def factory(
         log.info(f"Waiting for {waiting_time}sec to generate msg")
         time.sleep(waiting_time)
 
-        # Check messages are sent and received
-        threads = amq.run_in_bg(
-            namespace=kafka_namespace, value=value, since_time=since_time
-        )
+        if validate_messages:
```

Comment:
what if only run_in_bg=True and validate_messages=False ? This condition doesnt seems to be correct

Reply:
run_in_bg has to be called only when validate_messages is set to True. Now the code will throw exception for this case